client/landing/gutenboarding/path.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { generatePath, useRouteMatch } from 'react-router-dom';
+import { getLanguageSlugs } from '../../lib/i18n-utils';
+import { ValuesType } from 'utility-types';
+
+export const Step = {
+	IntentGathering: 'about',
+	DesignSelection: 'design',
+	PageSelection: 'pages',
+	Signup: 'signup',
+	CreateSite: 'create-site',
+} as const;
+
+export const langs: string[] = getLanguageSlugs();
+export const steps = Object.keys( Step ).map( key => Step[ key as keyof typeof Step ] );
+
+export const path = `/:step(${ steps.join( '|' ) })?/:lang(${ langs.join( '|' ) })?`;
+
+export type StepType = ValuesType< typeof Step >;
+
+export function makePath( step: StepType | undefined, lang?: string ) {
```

Comment:
Do we need `makePath` at all?

Reply:
Currently `makePath` is used inside `window.AppBoot`. This can be eliminated if the react rendering part inside AppBoot is refactored out as a function component because the lint problem I'm getting is that hooks like `usePath()` needs to be called inside a function component.

Reply:
@sirreal I just made a commit demonstrating how it looks like to consolidate `makePath()` & `usePath()` and refactoring the `<Redirect>` component out from `AppBoot` into a separate `<RedirectFallback>` component in order to use `usePath()` hook.

See: https://github.com/Automattic/wp-calypso/pull/39312/commits/cf6ec5b8d99f0931a62d1d50af1a595e777c368d